jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -1164,10 +1168,27 @@ public Method getDeclaredMethod(String name, Class<?>... parameterTypes) throws
  */
 @CallerSensitive
 public Class<?> getDeclaringClass() {
-	Class<?> declaringClass = getDeclaringClassImpl();
+	if (cachedDeclaringClass == null) {
+		if (cachedDeclaringClassOffset == -1) {
+			try {
+				Field cachedDeclaringClassField = Class.class.getDeclaredField("cachedDeclaringClass"); //$NON-NLS-1$
+				cachedDeclaringClassOffset = getUnsafe().objectFieldOffset(cachedDeclaringClassField);
+			} catch (NoSuchFieldException e) {
+				throw newInternalError(e);
+			}
+		}
+		/*[IF Sidecar19-SE]
+		getUnsafe().putObjectRelease(this, cachedDeclaringClassOffset, getDeclaringClassImpl());
+		/*[ELSE]*/
+		getUnsafe().putOrderedObject(this, cachedDeclaringClassOffset, getDeclaringClassImpl());
+		/*[ENDIF]*/
+	}
+
+	Class<?> declaringClass = cachedDeclaringClass;
```

Comment:
Please follow the other existing examples in this class, and declare and assign the local before checking for null. Then assign the local variable to the value inside the if null block. The unsafe call ensures writes won't be reordered but doesn't guarantee reads. This applied to all three sets of changes.

```
Object localVar = cachedValue;
if (localVar == null) {
   localVar = <computed value>;
   unsafe.putObjectRelease(this, offset, localVar);
}
<use localVar>
```